pkg/logger/zerologr.go

```diff
@@ -0,0 +1,226 @@
+// Copyright 2019 Jorn Friedrich Dreyer
+// Modified 2021 Serhii Mikhno
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// package zerologr defines an implementation of the github.com/go-logr/logr
+// interfaces built on top of zerolog (github.com/rs/zerolog).
+
+package logger
+
+import (
+	"fmt"
+	"os"
+	"sync"
+
+	"github.com/go-logr/logr"
+	"github.com/rs/zerolog"
+)
+
+const (
+	debugVerbosity = 2
+	traceVerbosity = 8
+	timeFormat     = "2006-01-02 15:04:05.000"
+)
+
+var (
+	log Zerologr
+	mu  sync.RWMutex
+)
+
+// this prevents from using non-initialised logger
+func init() {
+	fixByFile := []string{"asm_amd64.s", "proc.go", "zerologr.go"}
```

Comment:
I know this was code you inherited but I'm curious if you know what the reason for these changes were.

Reply:
In current state it works when `Debugf()` is called from some function in buffer.go which is called from some test function in buffer_test.go. In this case logger isn't initialised and 
1) It prevent from null pointer exception
2) fixing function name and file name in log.  ( `...[zerologr.go] [Debugf]...` )

But if we do everything in a right way, and logger will be initialised in pkg itself (or even in test) - we don't need this hacky workaround at all.